tsearch.c.inc

```diff
@@ -52,20 +46,20 @@ static struct naive_tsearch_tree **naive_tsearch_traverse(const void *key, struc
 NAIVE_TSEARCH_API
 void *NAIVE_TSEARCH_TFIND(const void *key, void *const *rootp, naive_tsearch_cmp *compar)
 {
-    return *naive_tsearch_traverse(key, (void*)rootp, compar, 0, NULL);
+    return *naive_tsearch_traverse(key, (struct naive_tsearch_tree**)rootp, compar, 0, NULL);
 }
 
 NAIVE_TSEARCH_API
 void *NAIVE_TSEARCH_TSEARCH(const void *key, void **rootp, naive_tsearch_cmp *compar)
 {
-    return *naive_tsearch_traverse(key, (void*)rootp, compar, 1, NULL);
+    return *naive_tsearch_traverse(key, (struct naive_tsearch_tree**)rootp, compar, 1, NULL);
 }
 
 NAIVE_TSEARCH_API
-void *NAIVE_TSEARCH_TDELETE(const void *NAIVE_TSEARCH_RESTRICT key, void **NAIVE_TSEARCH_RESTRICT rootp, naive_tsearch_cmp *compar)
+void *NAIVE_TSEARCH_TDELETE(const void *__restrict key, void **__restrict rootp, naive_tsearch_cmp *compar)
```

Comment:
At first I was going to complain that `__restrict` is not standard, but since [Wikipedia](https://en.wikipedia.org/wiki/Restrict) says it is supported by GCC, Clang, and Visual C++, it is probably about as good as we can get for `restrict support`. No action here, just a comment.